hbase-server/src/test/java/org/apache/hadoop/hbase/mob/TestMobCompaction.java

```diff
@@ -0,0 +1,375 @@
+    Connection conn = ConnectionFactory.createConnection(this.conf);
+    this.admin = conn.getAdmin();
+    this.hcd = new HColumnDescriptor(fam);
+    this.hcd.setMobEnabled(true);
+    this.hcd.setMobThreshold(mobLen);
+    this.hcd.setMaxVersions(1);
+    this.hdt.addFamily(hcd);
+    if (admin.tableExists(hdt.getTableName())) {
+      admin.disableTable(hdt.getTableName());
+      admin.deleteTable(hdt.getTableName());
+    }
+    admin.createTable(hdt);
+    table = conn.getTable(hdt.getTableName());
+  }
+
+  private void printConf() {
+    LOG.info("To run stress test, please change HBase configuration as following:");
+    LOG.info("hfile.format.version=3");
+    LOG.info("hbase.master.hfilecleaner.ttl=0");
+    LOG.info("hbase.hregion.max.filesize=200000000");
+    LOG.info("hbase.client.retries.number=100");
+    LOG.info("hbase.hregion.memstore.flush.size=800000");
+    LOG.info("hbase.hstore.blockingStoreFiles=150");
+    LOG.info("hbase.hstore.compaction.throughput.lower.bound=50000000");
+    LOG.info("hbase.hstore.compaction.throughput.higher.bound=100000000");
+    LOG.info("hbase.master.mob.cleaner.period=0");
+    LOG.info("hbase.mob.default.compactor=org.apache.hadoop.hbase.mob.FaultyMobStoreCompactor");
+    LOG.warn("injected.fault.probability=x, where x is between 0. and 1.");
+
+  }
+
+  private HTableDescriptor createTableDescriptor(final String name, final int minVersions,
+      final int versions, final int ttl, KeepDeletedCells keepDeleted) {
+    HTableDescriptor htd = new HTableDescriptor(TableName.valueOf(name));
+    return htd;
+  }
+
+  private HTableDescriptor createTableDescriptor(final String name) {
+    return createTableDescriptor(name, HColumnDescriptor.DEFAULT_MIN_VERSIONS, 1,
+      HConstants.FOREVER, HColumnDescriptor.DEFAULT_KEEP_DELETED);
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    HTU = new HBaseTestingUtility();
+    hdt = HTU.createTableDescriptor("testMobCompactTable");
+    conf = HTU.getConfiguration();
+
+    initConf();
+
+    // HTU.getConfiguration().setInt("hbase.mob.compaction.chore.period", 0);
+    HTU.startMiniCluster();
+    admin = HTU.getAdmin();
+
+    hcd = new HColumnDescriptor(fam);
+    hcd.setMobEnabled(true);
+    hcd.setMobThreshold(mobLen);
+    hcd.setMaxVersions(1);
+    hdt.addFamily(hcd);
+    table = HTU.createTable(hdt, null);
+  }
+
+  private void initConf() {
+
+    conf.setInt("hfile.format.version", 3);
+    conf.setLong(TimeToLiveHFileCleaner.TTL_CONF_KEY, 0);
+    conf.setInt("hbase.client.retries.number", 100);
+    conf.setInt("hbase.hregion.max.filesize", 200000000);
+    conf.setInt("hbase.hregion.memstore.flush.size", 800000);
+    conf.setInt("hbase.hstore.blockingStoreFiles", 150);
+    conf.setInt("hbase.hstore.compaction.throughput.lower.bound", 52428800);
+    conf.setInt("hbase.hstore.compaction.throughput.higher.bound", 2 * 52428800);
+    conf.setDouble("injected.fault.probability", failureProb);
```

Comment:
the ~4 places this gets used should be updated to better reflect the scope. e.g. `hbase.mob.compaction.fault.probability`.